nixos/modules/profiles/macos-builder.nix

```
@@ -103,6 +103,19 @@ in
     # server that QEMU provides (normally 10.0.2.3)
     networking.nameservers = [ "8.8.8.8" ];
 
+    # The linux builder is a lightweight VM for remote building; not evaluation.
+    nix.channel.enable = false;
+    # remote builder uses `nix-daemon` (ssh-ng:) or `nix-store --serve` (ssh:)
+    # --force: do not complain when missing
+    # TODO: install a store-only nix
+    #       https://github.com/NixOS/rfcs/blob/master/rfcs/0134-nix-store-layer.md#detailed-design
+    environment.extraSetup = ''
+      rm --force $out/bin/{nix-instantiate,nix-build,nix-shell,nix-prefetch*,nix}
+    '';
```

My `linux-builder` has gotten into a bad state and removing these commands makes it very hard to debug the situation

@roberth I wonder if it's better to revert this section

---

Which commands do you need to run?

Would an opt-in option be sufficient for your purposes? Something like:
```suggestion
    environment.extraSetup = mkIf (!cfg.nix.evaluator.enable) ''
      rm --force $out/bin/{nix-instantiate,nix-build,nix-shell,nix-prefetch*,nix}
    '';
```

Currently it's meant to simulate a store-only `nix`, so there's little benefit to this code as is, except manage expectations when we can reduce the linux builder's closure by only incorporating a store-only build of `nix`.

---

I want to run something like `nix build --rebuild` so ideally the `nix` command

If I'm understanding correctly, the default would be to include the `nix` command, which would be my preference as otherwise if your `linux-builder` gets in a bad state, it might be quite annoying/hard to bootstrap another one rather than be able to potentially fix it

---

> it might be quite annoying/hard to bootstrap another

I thought it was basically fully automated, except for clearing it.

Anyway, not being able to debug it properly is annoying, so I'm ok to revert this.

---

Even the clearing is automated, too (we set `min-free` and `max-free`), but it might still be worth restoring the `nix` command-line tools for debugging purposes

---

I was referring to bootstrapping my own Linux builder locally, but I realize now that if you just comment out any custom configuration (if you're using nix-darwin) and use a cached Linux builder from Nixpkgs that should work fine

I've a PR here: https://github.com/NixOS/nixpkgs/pull/347205